connect-examples/v2/node_gift-cards/public/js/amount-bar.js

```diff
@@ -87,8 +87,20 @@ class AmountBar {
    */
   setAmountChosen(amount = 0) {
     // We dont want rounding on our pay button, pass in false to our formatting function.
-    document.getElementById("pay-button").innerHTML = "Pay " + this.formatMoneyFunction(amount, this.currency, false);
-    document.getElementById("amount-bar__value").value = amount;
+    let payButton = document.getElementById("pay-button")
+    if (amount > this.maxAllowedToAdd) {
+      payButton.innerHTML = "Amount exceeds maximum balance allowed";
+
+      payButton.setAttribute("disabled-by-amount", "true");
+    } else {
+      payButton.innerHTML = "Pay " + this.formatMoneyFunction(amount, this.currency, false);
+      document.getElementById("amount-bar__value").value = amount;
+
+      payButton.setAttribute("disabled-by-amount", "false");
+    }
+
+    // HACK: control button in combination with logic from the select dropdown
```

Comment:
Let's remove the word hack

Reply:
what can we replace it with to indicate that this code is only here for a specific use case?

Reply:
How about:
`NOTE: This functionality is used specifically for coordinating logic between the dropdown and the pay button`

Reply:
I just think it's better than explicitly using the word hack

Reply:
You are the best!

Reply:
I didn't do anything, you did it all!